code/game/jobs/job_exp.dm

```diff
@@ -0,0 +1,197 @@
+	set category = "Admin"
+	set name = "Check Player Playtime"
+	if(!check_rights(R_ADMIN))
+		return
+	var/msg = "<html><head><title>Playtime Report</title></head><BR>Playtime:<BR>"
+	for(var/client/C in clients)
+		var/list/play_records = params2list(C.prefs.exp)
+		var/gen_exp_time = text2num(play_records["Crew"])
+		gen_exp_time = num2text(round(gen_exp_time / 60)) + "h"
+		msg += "<BR> - [key_name_admin(C.mob)]: <A href='?_src_=holder;getplaytimewindow=\ref[C.mob]'>" + C.get_exp_general() + "</a> "
+	if(msg != "")
+		src << browse(msg, "window=Player_playtime_check")
+	log_admin("[key_name(usr)] checked player playtime")
+	feedback_add_details("admin_verb","MCPP") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+
+/client/proc/cmd_show_exp_panel(var/client/C in clients)
+	if(!check_rights(R_ADMIN))
+		return
+	var/body = "<html><head><title>Playtime for [C.key]</title></head><BR>Playtime:"
+	body += C.get_exp_report()
+	body += "</HTML>"
+	src << browse(body, "window=playerplaytime;size=550x615")
+
+
+// Procs
+
+/proc/has_exp_for_job(mob/M, rank)
+	var/datum/job/job = job_master.GetJob(rank)
+	if(!job)
+		return 1
+	if(!job.exp_requirements || !job.exp_type)
+		return 1
+	if(!job_is_xp_locked(rank))
+		return 1
+	if(!M.client)
+		return 0
+	if(check_rights(R_ADMIN, 0, M))
+		return 1
+	var/list/play_records = params2list(M.client.prefs.exp)
+	var/isexempt = text2num(play_records["Exempt"])
+	if(isexempt)
+		return 1
+	var/my_exp = text2num(play_records[job.exp_type])
+	var/job_requirement = text2num(job.exp_requirements)
+	if(config.use_exp_restrictions_heads_hours && ((rank in command_positions) || rank == "AI"))
+		job_requirement = text2num(config.use_exp_restrictions_heads_hours) * 60
+	if(my_exp >= job_requirement)
+		return 1
+	else
+		return 0
+
+/proc/job_is_xp_locked(jobtitle)
+	if(!config.use_exp_restrictions_heads && ((jobtitle in command_positions) || jobtitle == "AI"))
+		return 0
+	if(!config.use_exp_restrictions_other && !((jobtitle in command_positions) || jobtitle == "AI"))
+		return 0
+	return 1
+
+
+/mob/proc/get_exp_report()
+	if(client)
+		return client.get_exp_report()
+	else
+		return "[src] has no client."
+
+/client/proc/get_exp_report()
+	if(!config.use_exp_tracking)
+		return "Tracking is disabled in the server configuration file."
+
+	var/list/play_records = params2list(prefs.exp)
+	if(!play_records.len)
+		return "[src] has no records."
+
+	var/return_text = ""
+
+	var/list/exp_data = list()
+	for(var/cat in exp_jobsmap)
+		if(text2num(play_records[cat]))
+			exp_data[cat] = text2num(play_records[cat])
+		else
+			exp_data[cat] = 0
+
+	for(var/dep in exp_data)
+		if(exp_data[dep] > 0)
+			if(dep == "Exempt")
+				return_text += "<BR>- Exempt (all jobs auto-unlocked)"
+			else
+				return_text += "<BR>- [dep] " + get_exp_format(exp_data[dep]) + " (" + num2text(round(exp_data[dep]/exp_data["Crew"]*100)) + "%)"
```

Comment:
At this point, there's no sanity check that played Crew time isn't 0. If it's 0 - for example, someone only has observer playtime so far - this will divide by 0 and runtime.

Also, if someone has very little crew playtime and a lot of observer playtime, I'm pretty sure this will show some massive %, since it seems to assume "Crew" encompasses all play time. There should probably be a category that increases any time any other category does.